### PR TITLE
Update Makefile

### PR DESCRIPTION
**Pull request by kamo-naoyuki**
- Change to remove chainer from requirements of setup.py and install it from Makefile
    - chainer=6.0.0 has incompatible requirement of protobuf to tensorboardX #2188 
      As workaround, I removed it from setup.py.
   - We still have conflicted packages due to underthesea or languageflow.
- I changed the default PyTorch to 1.4.0.
   - We should use pytorch=1.5.1, but espnet1 doesn't work, so I set 1.4.0 for now #2169
- I removed kaldi installation part from Makefile.
   - KALDI is mandatory argument: `$ make KALDI=<kaldi-path>`
- New variables:
    - CPU_ONLY: Perform on CPU mode
    - CONDA: You can specify existing anaconda path
    - CONDA_ENV_NAME
    - CHAINER_VERSION
    - USE_PIP: Use pip to install torch even if it's anaconda
- Romoved:
    - CUDA_VERSION: Detecting cuda version from `nvcc`
    - CUPY_VERSION: Use CHAINER_VERSION instead
- Makefile creates `tools/activate_python.sh`. `path.sh` invokes it.
    - To fix a current issue: If already in another anaconda env when make, it's failed to install packages because `conda deactivate` is not called.
    - Now we can easily replace python environment rewriting `tools/activate_python.sh`.
- Implement `tools/setup_cuda_env.sh`
    - A util to setup CUDA environment variables, $PATH, $LD_LIBRARY_PATH, etc.
    - This file is not called in Makefile.
- Rename `tools/install_*.sh` -> `tools/installers/install_*.sh`

### CODE REVIEW

File: doc/installation.md [hunk 1]
```diff
@@ -185,27 +182,47 @@ $ cd <espnet-root>/tools
...
+You can skip the installation of new environment by creating `activate_python.sh`.
+
+```sh
+$ cd <espnet-root>/tools
+$ rm -f activate_python.sh; touch activate_python.sh
+$ make KALDI=<kaldi-root> USE_PIP=0
+```
 
-To install in a terminal that does not have a GPU installed, just clear the version of `CUPY` as follows:
+If your Python is anaconda, you don't need to give `USE_PIP`.
```
Comment (sw005320) [inline doc/installation.md:202]:
```suggestion
If your Python is anaconda, you don't need to provide `USE_PIP`.
```

File: setup.py [hunk 2]
```diff
@@ -75,22 +75,25 @@
     ],
 }
 try:
+    # NOTE(kamo): There packages are not listed if installing from pypi server
```
Comment (sw005320) [inline setup.py:78]:
```suggestion
    # NOTE(kamo): These packages are not listed if installing from the PyPI server
```
??? Could you check it?

**Comment by kamo-naoyuki** [comment]
@sw005320 Sorry, I'll consider editor setting. After ci, I'll merge

**Comment by sw005320** [comment]
> @sw005320 Sorry, I'll consider editor setting. After ci, I'll merge

No problem at all.
Better documentation is important teamwork. I'll try to help as much as possible.

**Pull request merged**